my-app/src/modules/Cart/ListCartProducts.jsx

```diff
@@ -0,0 +1,79 @@
+import React, {useContext} from "react";
+import { Link } from "react-router-dom";
+import useAllCartItems from "./Services/ListItemsApi";
+import handleDelete from "./Services/DeleteApi";
+import { MyContext } from '../../data/MyContext';
+
+export default function ListCartProducts ({ updateTotalAmount }) {
+  const { cartItems, removeCartItem } = useAllCartItems();
+  const { productQuantities, setProductQuantities } = useContext(MyContext);
+
+  if(cartItems.length > 0)
+  {
+     const calculateTotal = () => {
+      return cartItems.reduce((acc, cartItem) => acc + cartItem.subtotal, 0);
+    };
+
+    const newTotal = calculateTotal();
+    updateTotalAmount(newTotal);
+  }
+
+  const onDeleteClick = async (cartItem) => {
+    try {
+      await handleDelete(cartItem.product.id);
```

Comment:
Also, Use sync await inside services.js
```suggestion
    removeFromCart(cartItem.product.id);
```